### PR TITLE
ZEPPELIN-3072: Zeppelin UI becomes slow/unresponsive if there are too many notebooks

### PR DESCRIPTION
**Pull request by prabhjyotsingh**
### What is this PR for?
Zeppelin UI becomes slow/unresponsive if there are too many notebooks

Have attached a notebook directory in [JIRA](https://issues.apache.org/jira/secure/attachment/12898650/notebook.zip) with 500+ notebooks, now with these notebooks, every time user goes to homepage Zeppelin UI becomes unresponsive for few seconds.

### What type of PR is it?
[Bug Fix | Improvement]

### Todos
* [x] - Fix search box
* [x] - Order of notebook

### What is the Jira issue?
* [ZEPPELIN-3072](https://issues.apache.org/jira/browse/ZEPPELIN-3072)

### How should this be tested?
 Create 500+ notebook, or import it from [JIRA](https://issues.apache.org/jira/secure/attachment/12898650/notebook.zip), now observe UI it becomes slow/laggy while homepage is rendering. 

### Screenshots (if appropriate)

Before:
before

After:
after

### Questions:
* Does the licenses files need update? N/A
* Is there breaking changes for older versions? N/A
* Does this needs documentation? N/A

### CODE REVIEW

**Comment by tinkoff-dwh** [comment]
navbar have the same tree

**Comment by prabhjyotsingh** [comment]
@tinkoff-dwh Sure let me work on that as well.

**Comment by tinkoff-dwh** [comment]
@prabhjyotsingh 
maybe more pretty http://sroze.github.io/ngInfiniteScroll/ . or a another implementation of infinite scrolling

**Comment by prabhjyotsingh** [comment]
Yes, thanks I'm trying to explore infinity scroll.

**Comment by prabhjyotsingh** [comment]
@tinkoff-dwh Thank you for the link. This seems to perform well.

Preview GIF
after


Let me know your thoughts.

**Comment by tinkoff-dwh** [comment]
works fine! But have some problems with sorting.

sorting on ui 
1) folders (alphabet)
2) note (alphabet)

load home page, we have part of tree with "local sorting" (one folder on top Zeppelin Tutorial)


after scroll to bootom and return to top

**Comment by prabhjyotsingh** [comment]
Sure, in my todo list. Will find this, if this implementation/UX looks OK

**Comment by tinkoff-dwh** [comment]
i think looks good

**Comment by prabhjyotsingh** [comment]
Sure, I was experimenting with few more infinity scroll, let me try those for performance, then will fix this.

File: zeppelin-web/src/components/navbar/navbar.html [hunk 1]
```diff
@@ -46,13 +46,13 @@
             <li class="divider hidden-xs"></li>
             <div id="notebook-list" class="scrollbar-container" ng-if="isDrawNavbarNoteList">
               <li class="filter-names" ng-include="'components/note-name-filter/note-name-filter.html'"></li>
-              <div ng-if="!query.q || query.q === ''">
-              <li ng-repeat="node in navbar.notes.root.children | orderBy:node:false:navbar.arrayOrderingSrv.noteComparator track by node.id"
```
Comment (prabhjyotsingh) [inline zeppelin-web/src/components/navbar/navbar.html]: Moved this order by to note-list.factory.js

**Comment by tinkoff-dwh** [comment]
Tested. LGTM

found bug for navbar dropdown (bug into branch master). 
1) Open navbar list of notes 
2) click to some folder
3) click to note in folder
4) Note loaded but dropdown menu not hide

**Comment by prabhjyotsingh** [comment]
@tinkoff-dwh Look's like `expand-collapse.directive.js:44`  -> "event.stopPropagation()" was prevening it to close, have fixed it.

**Comment by tinkoff-dwh** [comment]
yep, all works now

**Comment by prabhjyotsingh** [comment]
CI green; will merge this if no more discussion.

**Comment by tinkoff-dwh** [comment]
@prabhjyotsingh 
https://github.com/sroze/ngInfiniteScroll/blob/master/LICENSE

**Comment by prabhjyotsingh** [comment]
Sure, thanks a lot let me add that.

**Comment by tinkoff-dwh** [comment]
need to add in https://github.com/apache/zeppelin/blob/master/LICENSE

**Comment by prabhjyotsingh** [comment]
Sure yes, just missed it, have updated it.

**Pull request closed without merging**